smartapp-core/src/main/java/com/smartthings/sdk/smartapp/core/extensions/contextstore/DefaultInstalledAppContext.java

```diff
@@ -0,0 +1,148 @@
+package com.smartthings.sdk.smartapp.core.extensions.contextstore;
+
+import java.time.Instant;
+
+import com.smartthings.sdk.smartapp.core.models.InstallData;
+import com.smartthings.sdk.smartapp.core.models.InstalledApp;
+import com.smartthings.sdk.smartapp.core.models.UpdateData;
+
+
+/**
+ * This is the default implementation of InstalledAppContext which stores
+ * basic information related to installed apps. In particular, this includes
+ * token information to allow for keeping them up-to-date for out of band
+ * API requests.
+ */
+public class DefaultInstalledAppContext implements InstalledAppContext {
+    private InstalledApp installedApp;
+    private String authToken;
+    private Instant authTokenExpiration;
+    private String refreshToken;
+    private Instant refreshTokenExpiration;
+
+    public static DefaultInstalledAppContext from(InstallData installData) {
+        return new DefaultInstalledAppContext(installData.getInstalledApp(), installData.getAuthToken(),
+                installData.getRefreshToken());
+    }
+
+    public static DefaultInstalledAppContext from(UpdateData updateData) {
```

Comment:
Does `context` work on Events as well?

Reply:
I should document these methods. They are convenience methods used _by the context store_ when the app is installed or updated to create the initial instance to persist. The idea is that any code can use the context (getting it via the context store's `get` method) but nobody except the context store itself should be instantiating them.